cardano-api/src/Cardano/Api/Typed.hs

```diff
@@ -3105,19 +3114,18 @@ instance SerialiseAsRawBytes (Hash PaymentKey) where
       PaymentKeyHash . Shelley.KeyHash <$> Crypto.hashFromBytes bs
 
 instance HasTextEnvelope (VerificationKey PaymentKey) where
-    textEnvelopeType _ = "PaymentVerificationKeyShelley"
-    -- TODO: include the actual crypto algorithm name, to catch changes:
-{-
+    textEnvelopeType _ = "PaymentVerificationKeyShelley_"
                       <> fromString (Crypto.algorithmNameDSIGN proxy)
       where
         proxy :: Proxy (Shelley.DSIGN ShelleyCrypto)
         proxy = Proxy
--}
 
 instance HasTextEnvelope (SigningKey PaymentKey) where
-    textEnvelopeType _ = "SigningKeyShelley"
-    -- TODO: include the actual crypto algorithm name, to catch changes
-    -- TODO: fix these inconsistent names for the public testnet re-spin
+    textEnvelopeType _ = "PaymentSigningKeyShelley_"
+                      <> fromString (Crypto.algorithmNameDSIGN proxy)
+      where
+        proxy :: Proxy (Shelley.DSIGN ShelleyCrypto)
+        proxy = Proxy
```

Comment:
Ideally, I would have wanted to concat the strings without the `_` between them. However, some of the algorithm names are formatted differently to others (e.g. `ed25519`, `PraosVRF`), so I thought it'd look odd to see something like `PaymentSigningKeyShelleyed25519`.